cyences_app_for_splunk/default/data/ui/views/cs_network_reports.xml

```diff
@@ -57,56 +74,339 @@
+        <option name="cyences_app_for_splunk.network_telemetry_map.drilldownTokenValue">| tstats `cs_summariesonly_network_traffic` count, sum(All_Traffic.bytes) as total_bytes, sum(All_Traffic.packets) as total_packets from datamodel=Network_Traffic where `cs_private_ips(All_Traffic.dest_ip)` AND `cs_public_ips(All_Traffic.src_ip)` AND (All_Traffic.dest_ip=$tkn_ip$ OR All_Traffic.src_ip=$tkn_ip$) by All_Traffic.src_ip, All_Traffic.dest_ip, All_Traffic.dest_port, All_Traffic.action 
+| `drop_dm_object_name(All_Traffic)` 
+| append 
+    [| inputlookup cs_tenable_vuln 
+    | search (vul_state="open" OR vul_state="reopened") vul_port!=0 vul_severity IN ($tkn_vul_severity$) 
+    | fields tenable_uuid, vul_port 
+    | join tenable_uuid type=left 
+        [| inputlookup cs_tenable_inventory 
+        | table tenable_uuid, ip] 
+    | makemv ip delim="~~" 
+    | stats count by ip vul_port 
+    | rename ip as dest_ip vul_port as dest_port 
+    | eval is_vul = 1 ] 
+| append 
+    [| search `cs_qualys_vuln` `cs_qualys_timerange` STATUS="ACTIVE" PORT=* severity IN ($tkn_vul_severity$) 
+    | fields IP PORT 
+    | stats count by IP PORT 
+    | rename IP as dest_ip PORT as dest_port 
+    | eval is_vul = 1 ] 
+| eval single_field = src_ip."#".action."#".total_bytes."#".total_packets 
+| stats count, values(single_field) as single_field, values(is_vul) as is_vul by dest_ip dest_port 
+| search single_field="*" 
+| mvexpand single_field 
+| rex field=single_field "^(?&lt;src_ip&gt;[^#]+)#(?&lt;action&gt;[^#]+)#(?&lt;total_bytes&gt;[^#]+)#(?&lt;total_packets&gt;[^#]+)$" 
+| fillnull is_vul 
+| iplocation src_ip 
+| rename lat as start_lat, lon as start_lon, Country as src_country, City as src_city, Region as src_region 
+| eval end_lat=`cs_network_home_location_lat`, end_lon=`cs_network_home_location_lon`, dest_country=`cs_home_country` 
+| sort 0 -is_vul, -action, -total_bytes 
+| eval total_KB = round(total_bytes/1024, 2) 
+| table src_ip, src_country, src_city, src_region, dest_ip, dest_port, dest_country, action, is_vul, total_KB, total_packets, start_lat, start_lon, end_lat, end_lon</option>
+        <option name="cyences_app_for_splunk.network_telemetry_map.mapLatitude">30</option>
+        <option name="cyences_app_for_splunk.network_telemetry_map.mapLongitude">0</option>
+        <option name="cyences_app_for_splunk.network_telemetry_map.mapZoom">2</option>
         <option name="drilldown">none</option>
         <option name="refresh.display">progressbar</option>
       </viz>
       <html>
+      <style>
+      table, th, td {
+      	border: 1px solid; 
+        text-align: center;
+    
+      }
+      th, tr {
+          padding: 15px
+      }
+      </style>
+        <div>
+          <p>
+            <b>Description:</b> Displays a traffic visual for the all vulnerable and top twenty not vulnerable networks for each action (allowed and blocked) with their respective source and destination pathways.</p>
```

Comment:
How to read the chart?

| Color | Description |
| -------- | ------------ |
| Red | Inbound traffic on the **vulnerable ports** that are **allowed** by the firewall. |
| Orange | Inbound traffic on the **non-vulnerable ports** that are **blocked** by the firewall. |
| Blue | Other Inbound traffic is **blocked** by the firewall. (Only top 20) |
| Green | Other Inbound traffic **allowed** by the firewall. (Only top 20) |

Vulnerable Port - If there is at least one open vulnerability with severity selected by the user in the dashboard filter on the device on that specific port (detected by vulnerability scanners).